v-next.sql

```diff
@@ -0,0 +1,493 @@
+					UPDATE openchpl.test_standard
+					SET deleted = true, last_modified_user = -1, certification_edition_id = 1
+					WHERE test_standard_id = userEnteredTestStandard.test_standard_id;
+				WHEN 1 THEN
+					
+					-- look up which edition it was
+					SELECT INTO referencedByEdition result
+					FROM 
+					(SELECT DISTINCT cp.certification_edition_id as result
+					FROM openchpl.certified_product cp
+					JOIN openchpl.certification_result cert ON cp.certified_product_id = cert.certified_product_id
+					JOIN openchpl.certification_result_test_standard certTestStd ON cert.certification_result_id = certTestStd.certification_result_id
+					JOIN openchpl.test_standard testStd on certTestStd.test_standard_id = testStd.test_standard_id
+					WHERE testStd.test_standard_id = userEnteredTestStandard.test_standard_id) innerQuery;
+					
+					raise notice 'Only listings for edition id % use user-entered test standard %', referencedByEdition, userEnteredTestStandard.number;
+			
+					-- set this test standard to this edition
+					UPDATE openchpl.test_standard
+					SET certification_edition_id = referencedByEdition, last_modified_user = -1
+					WHERE test_standard_id = userEnteredTestStandard.test_standard_id;
+				WHEN 2 THEN
+					raise notice 'Listings with both 2014 and 2015 use user-entered test standard %', userEnteredTestStandard.number;
+
+					-- set the existing test standard to 2014
+					UPDATE openchpl.test_standard
+					SET certification_edition_id = 2, last_modified_user = -1
+					WHERE test_standard_id = userEnteredTestStandard.test_standard_id;
+					
+					-- create a new duplicate test standard for 2015
+					INSERT INTO openchpl.test_standard(number, name, certification_edition_id, last_modified_user) VALUES
+					(userEnteredTestStandard.number, userEnteredTestStandard.name, 3, -1);
+					
+					-- find 2015 listings now pointing to the 2014 standard and make them point to the new 2015 standard
+					FOR mismatchTestStandardId IN 
+					(SELECT certTestStd.certification_result_test_standard_id as result
+					FROM openchpl.certified_product cp
+					JOIN openchpl.certification_result cert ON cp.certified_product_id = cert.certified_product_id
+					JOIN openchpl.certification_result_test_standard certTestStd ON cert.certification_result_id = certTestStd.certification_result_id
+					JOIN openchpl.test_standard testStd on certTestStd.test_standard_id = testStd.test_standard_id
+					WHERE testStd.test_standard_id = userEnteredTestStandard.test_standard_id
+					AND cp.certification_edition_id = 3)
+					LOOP
+						UPDATE openchpl.certification_result_test_standard
+						SET test_standard_id = 
+							(SELECT test_standard_id 
+							FROM openchpl.test_standard 
+							WHERE number = userEnteredTestStandard.number AND certification_edition_id = 3)
+						WHERE certification_result_test_standard_id = mismatchTestStandardId;
+					END LOOP;
+				ELSE
```

Comment:
How can this happen? If it was Case 0, then it was handled above. Same with 1 and 2. Is this only possible if it was actually Case 3? If it's actually something other than 0, 1, or 2, I feel like the error message should be different